src/render/canvas/canvas-renderer.ts

```diff
@@ -699,11 +721,71 @@ export class CanvasRenderer {
         let side = 0;
         for (const border of borders) {
             if (border.style !== BORDER_STYLE.NONE && !isTransparent(border.color)) {
-                await this.renderBorder(border.color, side++, paint.curves);
+                if (border.style === BORDER_STYLE.DASHED) {
+                    await this.renderDashedBorder(border.color, side++, paint.curves);
+                } else if (border.style === BORDER_STYLE.DOTTED) {
+                    await this.renderDottedBorder(border.color, side++, paint.curves);
+                } else if (border.style === BORDER_STYLE.DOUBLE) {
+                    await this.renderDoubleBorder(border.color, side++, paint.curves);
+                } else {
+                    await this.renderSolidBorder(border.color, side++, paint.curves);
+                }
```

Comment:
Consider Specifying State of SOLID otherwise iterating over the side akin to a "State Machine"

I tested the following locally, which handled fallback iteration of side. 

```javascript
for (const border of borders) {
            if (border.style !== BORDER_STYLE.NONE && !isTransparent(border.color)) {
                if (border.style === BORDER_STYLE.DASHED) {
                    await this.renderDashedBorder(border.color, side++, paint.curves);
                } else if (border.style === BORDER_STYLE.DOTTED) {
                    await this.renderDottedBorder(border.color, side++, paint.curves);
                } else if (border.style === BORDER_STYLE.DOUBLE) {
                    await this.renderDoubleBorder(border.color, side++, paint.curves);
                } else if (border.style === BORDER_STYLE.SOLID){
                    await this.renderSolidBorder(border.color, side++, paint.curves);
                } else {
                    side++
                }
            } else {
                side++
            }
        }
```